keras_hub/src/models/differential_binarization/losses.py

```diff
@@ -0,0 +1,126 @@
+        loss = 1 - 2.0 * intersection / union
+        return loss
+
+
+class MaskL1Loss:
+    def __init__(self, **kwargs):
+        pass
+
+    def __call__(self, y_true, y_pred, mask):
+        mask_sum = ops.sum(mask)
+        loss = ops.where(
+            mask_sum == 0.0,
+            0.0,
+            ops.sum(ops.absolute(y_pred - y_true) * mask) / mask_sum,
+        )
+        return loss
+
+
+class BalanceCrossEntropyLoss:
+    def __init__(self, negative_ratio=3.0, eps=1e-6, **kwargs):
+        self.negative_ratio = negative_ratio
+        self.eps = eps
+
+    def __call__(self, y_true, y_pred, mask, return_origin=False):
+        positive = ops.cast((y_true > 0.5) & ops.cast(mask, "bool"), "uint8")
+        negative = ops.cast((y_true < 0.5) & ops.cast(mask, "bool"), "uint8")
+        positive_count = ops.sum(ops.cast(positive, "int32"))
+        negative_count = ops.sum(ops.cast(negative, "int32"))
+        negative_count_max = ops.cast(
+            ops.cast(positive_count, "float32") * self.negative_ratio, "int32"
+        )
+
+        negative_count = ops.where(
+            negative_count > negative_count_max,
+            negative_count_max,
+            negative_count,
+        )
+        # Keras' losses reduce some axis. Since we don't want that here, we add
+        # a dummy dimension to y_true and y_pred
+        loss = keras.losses.BinaryCrossentropy(
+            from_logits=False,
+            label_smoothing=0.0,
+            axis=-1,
+            reduction=None,
+        )(y_true=y_true[..., None], y_pred=y_pred[..., None])
+
+        positive_loss = loss * ops.cast(positive, "float32")
+        negative_loss = loss * ops.cast(negative, "float32")
+
+        # hard negative mining, as suggested in
+        # [Real-time Scene Text Detection with Differentiable Binarization](https://arxiv.org/abs/1911.08947):
+        # Compute the threshold for hard negatives, and zero-out
+        # negative losses below the threshold. using this approach,
+        # we achieve efficient computation on GPUs
+
+        # compute negative_count relative to the element count of y_pred
+        negative_count_rel = ops.cast(negative_count, "float32") / ops.prod(
+            ops.cast(ops.shape(y_pred), "float32")
+        )
+        # compute the threshold value for negative losses and zero neg. loss
+        # values below this threshold
+        negative_loss_thresh = ops.quantile(
+            negative_loss, 1.0 - negative_count_rel
+        )
+        negative_loss = negative_loss * ops.cast(
+            negative_loss > negative_loss_thresh, "float32"
+        )
+
+        balance_loss = (ops.sum(positive_loss) + ops.sum(negative_loss)) / (
+            ops.cast(positive_count + negative_count, "float32") + self.eps
+        )
+
+        if return_origin:
+            return balance_loss, loss
+        return balance_loss
+
+
+class DBLoss(keras.losses.Loss):
```

Comment:
Again lets be consistent in our abbreviations. We have DiffBin, DB and DifferentialBinarization. Maybe DiffBin is the right middle ground everywhere? But as I mentioned above, not sure what is common in other projects.

Reply:
(see above comment) I'll rename as soon as we have agreed on a good name.